public/app/components/fontViewer/fontViewer.js

```diff
@@ -0,0 +1,477 @@
+      var fragment = document.createDocumentFragment();
+      var numPages = Math.ceil(vm.font.numGlyphs / cellCount);
+      for (var i = 0; i < numPages; i++) {
+        var link = document.createElement('span');
+        var lastIndex = Math.min(
+          vm.font.numGlyphs - 1, (i + 1) * cellCount - 1
+        );
+        link.textContent = i * cellCount + '-' + lastIndex;
+        link.id = 'p' + i;
+        link.addEventListener('click', pageSelect, false);
+        fragment.appendChild(link);
+        // A white space allows to break very long lines into multiple lines.
+        // This is needed for fonts with thousands of glyphs.
+        fragment.appendChild(document.createTextNode(' '));
+      }
+      pagination.appendChild(fragment);
+
+      initGlyphDisplay();
+      displayGlyphPage(0);
+
+      var glyphRange = vm.font.numGlyphs > 100 ? 100 : vm.font.numGlyphs;
+      selectedGlyphIndex = getRandomIntInclusive(0, glyphRange);
+      selectCurrentGlyph(selectedGlyphIndex);
+
+      displayGlyph(selectedGlyphIndex);
+      displayGlyphData(selectedGlyphIndex);
+    }
+
+    function getRandomIntInclusive(min, max) {
+      do {
+        var index = Math.floor(Math.random() * (max - min + 1)) + min;
+        var tempGlyph = vm.font.glyphs.get(index);
+      }
+      while (ignoreChars.indexOf(tempGlyph.name) != -1);
+      return index;
+    }
+
+    function cellSelect(event) {
+      if (!vm.font) { return; }
+      var firstGlyphIndex = pageSelected * cellCount;
+      var cellIndex = +event.target.id.substr(1);
+      var glyphIndex = firstGlyphIndex + cellIndex;
+
+      selectCurrentGlyph(glyphIndex);
+      if (glyphIndex < vm.font.numGlyphs) {
+        displayGlyph(glyphIndex);
+        displayGlyphData(glyphIndex);
+      }
+    }
+
+    function selectCurrentGlyph(index) {
+      //we use mod100 in order to get the right index(0-99) after the 1st page
+      index %= 100;
+      selectedGlyphIndex = index;
+      selectedGlyphPageIndex = currentPage;
+      deselectCurrentGlyph();
+      var selectedGlyph = document.getElementById('g' + index);
+      selectedGlyph.className += ' selected-glyph';
+    }
+
+    function deselectCurrentGlyph() {
+      var glyphToDeselect = document.getElementsByClassName('selected-glyph');
+      if (glyphToDeselect[0]) {
+        glyphToDeselect[0].className = 'item';
+      }
+    }
+
+    function prepareGlyphList() {
+      var marker = document.getElementById('glyph-list-end');
+      var parent = marker.parentElement;
+
+      for (var i = 0; i < cellCount; i++) {
+        var canvas = document.createElement('canvas');
+        canvas.width = cellWidth;
+        canvas.height = cellHeight;
+        canvas.className = 'item';
+        canvas.id = 'g' + i;
+        canvas.addEventListener('click', cellSelect, false);
+        enableHighDPICanvas(canvas);
+        parent.insertBefore(canvas, marker);
+      }
+    }
+
+    vm.drawPreviewText = function(textToRender) {
```

Comment:
Please order all `function`s beneath `vm.<somthing>`.

Also, running functions should go above function declarations.

Basically, as close as you can get this to the new formatted `codeEditor.js`.

I want to format all files based on [johnpapa's angular-styleguide](https://github.com/johnpapa/angular-styleguide/blob/master/a1/README.md).

This is already sooooo close to that. So I think it will be worth the little changes.

Basically, it's like so:

1. private `var` variables
2. hook `vm.functionName` to `function functionName()` with `vm.functionName = functionName`. (This way, it's easy to see without any inside logics, which functions are **public**)
3. All function activation (render \ init \ whatever)
4. then, all the function logic is at the end, with `function functionName` syntax.

Look for the **Bindable Members Up Top** section in the styleguide:
![image](https://cloud.githubusercontent.com/assets/10427304/17084750/a9652c80-51ce-11e6-9a87-55f04a89ff7d.png)